libcontainer/cgroups/devices/systemd.go

```diff
@@ -133,8 +134,21 @@ func systemdProperties(r *configs.Resources) ([]systemdDbus.Property, error) {
 			// rules separately to systemd) we can safely skip entries that don't
 			// have a corresponding path.
 			if _, err := os.Stat(entry.Path); err != nil {
-				logrus.Debugf("skipping device %s for systemd: %s", entry.Path, err)
-				continue
+				// Also check /sys/dev so that we don't depend on /dev/{block,char}
+				// being populated. (/dev/{block,char} is populated by udev, which
+				// isn't strictly required for systemd). Ironically, this happens most
+				// easily when starting containerd within a runc created container
+				// itself.
+				testPath, err := securejoin.SecureJoin("/sys", entry.Path)
+				if err != nil {
+					logrus.Errorf("error joining entry path: %s", err)
+					continue
```

Comment:
If that's an error, I guess we should not continue but return it.

Reply:
Or maybe we should not use securejoin here, since this is merely a check that such device exists, and we do not use `testPath` in any other way. Worst case scenario, a bad guy will trick the system into adding a line for a device that does not exist into systemd unit properties.

@cyphar WDYT?

Reply:
The original patch didn't have secure join, but I saw @AkihiroSuda preferred having it (not sure if it's _really_ needed here indeed); see https://github.com/opencontainers/runc/pull/3620#discussion_r985375647

Reply:
@AkihiroSuda WDYT?

Reply:
SGTM, but probably the reason not to use securejoin should be recorded in the comment lines in the source

Reply:
No problem, I removed securejoin and added a comment why it's not used.